src/networking.c

```diff
@@ -1470,6 +1478,24 @@ void freeClientAsync(client *c) {
     pthread_mutex_unlock(&async_free_queue_mutex);
 }
 
+/* Perform processing of the client before moving on to processing the next client
+ * this is useful for performing operations that affect the global state but can't
+ * wait until we're done with all clients. In other words can't wait until beforeSleep()
+ * return C_ERR in case client is no longer valid after call. */
+int beforeNextClient(client *c) {
+    /* Skip the client processing if we're in an IO thread, in that case we'll perform
+       this operation later in the fan-in stage of the threading mechanism */
+    if (io_threads_op != IO_THREADS_OP_IDLE)
+        return C_OK;
```

Comment:
we call this (beforeNextClient) function again there, right?
so if someone adds more jobs in this this (beforeNextClient) function, it'll automatically be covered?

```suggestion
    /* Skip the client processing if we're in an IO thread, in that case we'll perform
       this operation later (this function is called again) in the fan-in stage of the threading mechanism */
    if (io_threads_op != IO_THREADS_OP_IDLE)
        return C_OK;
```

Reply:
> we call this (beforeNextClient) function again there, right?
so if someone adds more jobs in this this (beforeNextClient) function, it'll automatically be covered?

correct